src/character.cpp

```diff
@@ -4920,26 +4920,37 @@ int Character::weariness_level() const
     int threshold = weary_threshold();
     int level = 0;
     amount -= threshold * get_option<float>( "WEARY_INITIAL_STEP" );
-    while( amount >= 0 ) {
-        amount -= threshold;
-        if( threshold > 20 ) {
-            threshold *= get_option<float>( "WEARY_THRESH_SCALING" );
+    // failsafe if threshold is zero; see #72242
+    if( threshold == 0 ) {
+        return level;
```

Comment:
The closer to 0 metabolism is the closer to 0 threshold is which means as you approach zero this function returns increasingly higher levels. Suddenly dropping that to 0 instead of whatever the max is when you finally hit 0 seems like the opposite of the trend.

Reply:
it is indeed not correct, but it is not something i made, and i clearly have no idea how to resolve it, so i just assume it is a feature and character with no metabolism should not be tired